Add request saving feature

steffenix opened the pull request:
The idea is to be able to make requests from using this endpoint and being able to get those requests afterward something like [requestb.in](http://requestb.in).

My use case would be to test API that require API keys that changes overtime (like APNS and GCM) and being able to make sure that the request has been done with the correct parameters.

edgurgel commented:
It's an interesting feature but  I would propose some changes:

Change `/store` and `retreive` to be `/store/:id` so that multiple users can send different ids and retrieve different ids

Then we can have 1 ETS table holding all the requests and each tuple is 

`{ id, request }`

And they can have more than 1 with `id`

Then if `retrieve/id` is requested all these tuples are removed, aggregated and sent.

We should also have some way to TTL the tuples of an id when they get too old. So that the ETS table does not consume too much memory?

What do you think?

steffenix commented:
The idea of having an id for each user is great, I din't think about it because I use this for local testing but it's definitely a feature needed for the heroku hosted version.

Would you be ok if we add `ecache`as a dependency to resolve the TTL part?

edgurgel commented:
What about https://github.com/sasa1977/con_cache ? I never used ecache but if you have experience maybe it's good enough?

steffenix commented:
Can't get why this test does not pass, would you give me an hint @edgurgel

edgurgel commented on test/retrieve_request_handler_tests.exs:
```diff
@@ -0,0 +1,17 @@
+  import HTTParrot.RetrieveRequestHandler
+
+  setup do
+    HTTParrot.RequestStore.clear(:test)
```
Because we are using meck, you need to add this to the setup:

`on_exit fn -> unload end`

edgurgel commented:
Cool after pulling your branch I noticed that we test on travis using:

`mix test --no-start` (.travis.yml)

We need to remove this `--no-start` then con_cache will be up an running 👍

edgurgel commented on lib/httparrot/request_store.ex:
```diff
@@ -0,0 +1,20 @@
+defmodule HTTParrot.RequestStore do
+
```
I wonder if you could add a `@moduledoc` to the module, @doc and @spec to the functions?

edgurgel replied:
Shall we also add tests just for the request store?

steffenix replied:
We could be we could also think that this is tested by the HTTPs tests and not write any as you wish.

steffenix commented:
I have added the doc fixed the test and added test on the RequestStore.
I had to update the elixir 1.2 version because of a bug with OTP 19 support.
https://github.com/elixir-lang/elixir/issues/4617

Please review.

edgurgel commented:
Thanks @steffenix! I will release a new version soon! 🎉
